src/topology/mod.rs

```diff
@@ -88,6 +88,12 @@ pub fn validate(config: &Config, exec: runtime::TaskExecutor) -> Option<Pieces>
 }
 
 impl RunningTopology {
+    /// Sends the shutdown signal to all sources and returns a future that resolves
+    /// once all components (sources, transforms, and sinks) have finished shutting down.
+    /// Transforms and sinks should shut down automatically once their input tasks finish.
+    /// Note that this takes ownership of `self`, so once this function returns everything in the
+    /// RunningTopology instance has been dropped except for the `tasks` map, which gets moved
+    /// into the returned future and is used to poll for when the tasks have completed.
```

Comment:
Would be good to note here that once the returned future is dropped everything then is fully dropped.

Reply:
Done